pkg/app/piped/diff/diff.go

```diff
@@ -258,6 +283,20 @@ func isNumberValue(v reflect.Value) bool {
 	}
 }
 
+func convertToNumber(v reflect.Value) (reflect.Value, bool) {
+	switch v.Kind() {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Float32, reflect.Float64:
+		return v, true
+	case reflect.String:
```

Comment:
looks like the value v which passed here is pass `isNumberValue` check 👀 And only value of kind (reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Float32, reflect.Float64) would pass that check ref: [L277](https://github.com/pipe-cd/pipe/pull/1875/files#diff-5e3f13f1f29b52a49818fb71df2b1a23fed001831c0b5f4f14a7f2fbc24de8f7R277).
I mean, looks like this `reflect.String` convert to number would never hit 🤔

Reply:
I was wrong, the `isNumberValue` and `convertToNumber` used for 2 different values 🙏